### PR TITLE
Cop 12161/rework tasklist flow (CYSA)

### PR DESCRIPTION
**Pull request by bowerscodes-COP**
Rework TaskList flow so that a 'Check your saved answers' screen is displayed when resuming a saved form, per [Jira Ticket COP-12161](https://support.cop.homeoffice.gov.uk/browse/COP-12161).

### CODE REVIEW

File: src/components/CheckYourAnswers/CheckYourAnswers.jsx [hunk 1]
```diff
@@ -60,7 +62,17 @@ const CheckYourAnswers = ({
...
+      const indexOfCurrentPage = pagesWithRows.map(object => object.id).indexOf(currentPage);
+      if (indexOfCurrentPage === 0) {
+        setPages(pagesWithRows.slice(0, 1));
+      } else {
+        setPages(pagesWithRows.slice(0, (indexOfCurrentPage)));
+      }
+    } else {
+      setPages(pagesWithRows);
+    };
   }, [_pages, onRowAction, setPages]);
```
Comment (fisherrj2) [inline src/components/CheckYourAnswers/CheckYourAnswers.jsx:76]: Looking at the GitHub Actions you will need to update the dependency array.  This in turn could affect behaviour so be careful to test again.

File: src/context/HooksContext/HooksContext.jsx [hunk 2]
```diff
@@ -6,7 +6,7 @@ const DEFAULT_HOOKS = {
   onFormComplete: () => {},
   onFormLoad: () => {},
   onGetComponent: (config, wrap) => null,
-  onPageChange: (pageId) => pageId,
+  onPageChange: (pageId, state) => pageId,
```
Comment (fisherrj2) [inline src/context/HooksContext/HooksContext.jsx:9]: Just wondered if you think it is necessary to add state here?  I wasn't sure how these default hooks are used and whether it was strictly necessary to add state as shown here.

File: src/components/CheckYourAnswers/CheckYourAnswers.jsx [hunk 3]
```diff
@@ -105,7 +117,7 @@ const CheckYourAnswers = ({
   return (
     <div className={DEFAULT_CLASS}>
       {title && !hide_title && (
-        <LargeHeading key='heading'>{title}</LargeHeading>
+        <LargeHeading key='heading'>{cysa ? 'Check your saved answers' : title}</LargeHeading>
```
Comment (fisherrj2) [inline src/components/CheckYourAnswers/CheckYourAnswers.jsx:120]: Is it worth adding this to a constant or even to the translation.json?

File: src/components/FormRenderer/FormRenderer.jsx [hunk 4]
```diff
@@ -185,15 +195,15 @@ const InternalFormRenderer = ({
     }
   }, [currentTask.fullPages, data, hubDetails?.sections, hubDetails?.nonSequential]);
 
-  const onPageChange = (newPageId) => {
+  const onPageChange = (newPageId, state) => {
     clearErrors();
     setPageId(newPageId);
-    hooks.onPageChange(newPageId);
+    hooks.onPageChange(newPageId, typeof state === 'undefined' ? {} : state);
```
Comment (fisherrj2) [inline src/components/FormRenderer/FormRenderer.jsx:201]: Is this check the best approach or should we ensure that state is always populated?

File: src/components/CheckYourAnswers/CheckYourAnswers.jsx (hunk 3 above)
Comment (bowerscodes-COP) [reply src/components/CheckYourAnswers/CheckYourAnswers.jsx]: Added as constant

**Review by fisherrj2 — APPROVED** [review]
Approved on the understanding that testing will be updated to ensure new features are covered.

**Pull request closed without merging**